source/matplot/util/popen.h

```diff
@@ -6,9 +6,18 @@
 #define MATPLOTPLUSPLUS_POPEN_H
 
 #ifdef _WIN32
+#include <io.h>
```

Comment:
Do we need these in the header file? Can't these just go to the cpp file?

Reply:
We can move `#include <io.h>` in the cpp file. However, `#include <stdio.h>` must be included in the header. I'm updating the PR following your comments.

Reply:
Does it work if we forward-declare `FILE` in the header file, then use `FILE*` in the header file, then include `stdio.h` in the cpp file?

Reply:
I woulnd't to this, to avoid adding a lot of `reinterpret_cast` in the implementation (because of the namespace). What do you think?

Reply:
Maybe I'm not miscommunicating. There's no `reinterpret_cast` in what I'm proposing.

popen.cpp:

```
#include <io.h>
```

becomes:

```
#include <io.h>
#include <stdio.h>
```

popen.h:

```
#include <stdio.h> //needed for FILE type
#include <windows.h>
```

becomes

```
#include <windows.h>
struct FILE;
```

but I just noticed this is incorrect for some other reason: The standard just states that FILE is 

> an object type capable of recording all the information needed to control a stream"; it's up to the implementation whether this is a typedef of a struct (whose name you don't know anyway), or something else.

In any case, it doesn't make much sense to avoid `stdio.h` when `windows.h`, which is a gazillion times heavier, is right next it.